src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -290,7 +306,13 @@ public function startApp(?\yii\log\Logger $logger = null): void
             unset($config['container']);
         }
 
-        $config = $this->mockMailer($config);
+        match ($this->mailMethod) {
+            self::MAIL_CATCH => $config= $this->mockMailer($config),
+            self::MAIL_EVENT_AFTER => $config['components']['mailer']['on ' . BaseMailer::EVENT_AFTER_SEND] = fn(MailEvent $event) => $this->emails[]  = $event->message,
```

Comment:
here you could now check for `MailEvent::$isSuccessful`

Reply:
What do we need to check?

Reply:
if I have some kind of behavior on mailer that filters out emails(by setting `MailEvent::$isValid` to false) then it shouldn't send this email, so this message should not go to `$this->emails` array